README.md

```diff
@@ -1,73 +1,320 @@
+| host                 | A host/IP to use for setting up a vpn connection. Only specified when overriding the name of hosts used by Ansible for SSH. | str | no   | - | left/right           |
+| public_key_src       | A path to a file on the controller host containing the public key used by this host for public key authentication without certificates.     | str    | no       | See [public_key fields](#public_key)| leftrsasigkey/rightrsasigkey |
+| public_key_content   | Contains the public key used by this host for public key authentication without certificates.                        | str         | no       | -                       | leftrsasigkey/rightrsasigkey |
+| cert_name            | Defines the certificate nickname of this host's certificate in the NSS database.                  | str         | no       | -                       | leftcert/rightcert   |
+| external             | Indicates that the associated host is external to the inventory (e.g. not managed by our inventory).    | bool        | no       | False                   | -                    |
+| subnets              | Defines a list of the subnets that should be available via the VPN connection.                    | list        | no       | -                       | leftsubnets/rightsubnets |
+
+#### public_key
+
+`public_key_src` specifies a path to a file on the controller host containing the public key used by this host for public key authentication without certificates. Otherwise, the user can directly specify the public key for this host by populating `public_key_content`. `public_key_content` can also accept a CKAID or nickname for a public key in the NSS database.
+
+Note that `public_key_src` and `public_key_content` may also be specified as host variables in the Ansible `all.hosts` section. The variable names in this case will be `vpn_public_key_src` and `vpn_public_key_content`. See [this example](#defining-public-keys-at-different-levels).
+
+**To be added later**: if neither `public_key_src` nor `public_key_content` is populated, the role will generate key pairs for each host.
+
+### Handling special cases with host names
+
+There are two ways to handle the special case when the names of the hosts in the Ansible inventory are not FQDN hostnames that you can use directly (e.g. they may be host aliases like "bastion_east"):  
+
+1. **Use the Ansible `hosts` configuration**
 
 ```yaml
-template_foo: "oof"
-template_bar: "baz"
+all:
+  hosts:
+    bastion_east:
+      ansible_host: bastion1.example.com
+      vpn_hostid: '@bastion-east'
+    bastion_west:
+      ansible_host: bastion2.example.com
+      vpn_hostid: '@bastion-west'
+    bastion_north:
+      ansible_host: bastion3.example.com
+      vpn_hostid: '@bastion-north'
+  vars:
+    vpn_connections:
+      - hosts:
+          bastion_east:
```

Comment:
It is not entirely clear what is used to tie the hosts names under `vpn_connections.hosts` to the inventory host names. From the expression below `hostvars[hostname].ansible_host` it seems that the hostname under `vpn_connections.hosts` references the hosts in inventory. But the example uses a `vpn_hostid` attribute, which is a reference back to the hostname. This may suggest to someone that the role would look up hosts in the inventory by locating those that have the `vpn_hostid` attribute equal to @_hostname_. If it is not the case, maybe a different attribute should be chosen as an example, or some other example values should be chosen, rather than those that seem to reference back the hostnames.

Reply:
What you described is expected behavior. The purpose of `vpn_hostid` is to provide an alternate name for a host (if, for example, the host has a long FQDN and the user instead wants to use a more friendly name when generating the vpn tunnel). You are correct that the example does not describe this clearly though. I'll make corrections to address that.

Reply:
And is `vpn_hostid` used as a reference into `vpn_onnections[...].hosts`?

Reply:
Yes.

Reply:
> Yes.

I don't think so.  `vpn_hostid` is a libreswan specific thing for the `leftid`/`righid` field.  Perhaps we should just omit `vpn_hostid` for now.